resilience4j-retry/src/main/java/io/github/resilience4j/retry/RetryConfig.java

```diff
@@ -52,6 +58,18 @@ public int getMaxAttempts() {
         return exceptionPredicate;
     }
 
+    public CheckedConsumer<Long> getSleepFunction() {
+        return sleepFunction;
+    }
+
+    public int getBufferSize() {
```

Comment:
I think I would prefer that we have to different Config classes for two different retry implementations. From the user perspective I might be overwhelmed by the configuration options even if they don't apply to the concrete Retry implementation. 
My suggestion: We should introduce a new `BudgetedRetryConfig` and a new `BudgetedRetry` interface and `BudgetedRetryImpl`. (Seperation of Concerns).

```` 
BudgetedRetryConfig config = BudgetedRetryConfig.custom()
                 .bufferSize(1000)
                 .retryThreshold(0.3)
                 .maxAttempts(3)
                 .build()
BudgetedRetry retry = BudgetedRetry.of("testName, config);
````

Reply:
I get the reason, but I think it is better to keep `Retry` interface and benefit from it. Retry Budget(RetryBudgetImpl) is just another implementation. Better documentation of `RetryConfig`?

Reply:
Since we have multiple implementations of Retry now, I don't like to static factory method `Retry.of()` in the interface anymore. How do you control which implementation do you want to use. Either by using different Config classes or we need a `RetryFactory`.
A better place for the static factory methods would have been the implementation class itself. :(

Reply:
@storozhukBM What do you think?

Reply:
Yep, we have similar issue with two implementations of RateLimiter.

Reply:
Ok, but this won't be backward compatible

Reply:
Yes we should first discuss pros and cons.